src/main/java/frc/team2412/robot/commands/diagnostic/DiagnosticRobotCommandGroup.java

```diff
@@ -0,0 +1,45 @@
+package frc.team2412.robot.commands.diagnostic;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.team2412.robot.subsystem.ClimbSubsystem;
+import frc.team2412.robot.subsystem.IndexSubsystem;
+import frc.team2412.robot.subsystem.IntakeSubsystem;
+import frc.team2412.robot.subsystem.ShooterSubsystem;
+
+public class DiagnosticRobotCommandGroup extends SequentialCommandGroup {
+    ShuffleboardTab tab = Shuffleboard.getTab("Self Diagnostic");
+    NetworkTableEntry intakeStatus, shooterStatus, indexStatus, climbStatus;
+
+    public DiagnosticRobotCommandGroup(
+            IntakeSubsystem intakeSubsystem,
+            ShooterSubsystem shooterSubsystem,
+            IndexSubsystem indexSubsystem,
+            ClimbSubsystem climbSubsystem) {
+        /*
+         * The Shuffleboard having issue of
+         * cannot properly display more than 3 rows of entry in one column
+         */
+        indexStatus = tab.add("Index Status", "Waiting").withPosition(0, 0).withSize(1, 1).getEntry();
+        shooterStatus = tab.add("Shooter Status", "Waiting").withPosition(1, 0).withSize(1, 1).getEntry();
+        intakeStatus = tab.add("Intake Status", "Waiting").withPosition(2, 0).withSize(1, 1).getEntry();
+        climbStatus = tab.add("Climb Status", "Waiting").withPosition(3, 0).withSize(1, 1).getEntry();
+        addRequirements(intakeSubsystem, shooterSubsystem, indexSubsystem, climbSubsystem);
+        addCommands(
+                new InstantCommand(() -> intakeStatus.setString("In Progress")),
+                new DiagnosticIntakeCommandGroup(intakeSubsystem),
+                new InstantCommand(() -> intakeStatus.setString("Finished")),
+                new InstantCommand(() -> shooterStatus.setString("In Progress")),
```

Comment:
Best to add a wait here instead of depending on the individual command groups to have waits